### PR TITLE
build: create build-parent-it pom to collect common integration tests set-up

### PR DESCRIPTION
**Pull request by lburgazzoli**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: tooling/create-extension-templates/parent-pom.xml [hunk 1]
```diff
@@ -23,9 +23,9 @@
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>org.apache.camel.quarkus</groupId>
-        <artifactId>camel-quarkus-build-parent</artifactId>
+        <artifactId>camel-quarkus-build-parent-it</artifactId>
         <version>[=version]</version>
-        <relativePath>../../poms/build-parent/pom.xml</relativePath>
+        <relativePath>../../poms/build-parent-it/pom.xml</relativePath>
```
Comment (ppalaga) [inline tooling/create-extension-templates/parent-pom.xml:28]: I think this should stay as is. This the template for the parent of runtime and deployment modules.

The template that needs change is `tooling/create-extension-templates/integration-test-pom.xml`. I think we can hardcode the artifactId `camel-quarkus-build-parent-it` there.

Comment (lburgazzoli) [reply tooling/create-extension-templates/parent-pom.xml:28]: oh sorry, I changed the wrong one, maybe having a structure similar to the the final directory layout could help a little

Comment (lburgazzoli) [reply tooling/create-extension-templates/parent-pom.xml]: @ppalaga done

Comment (ppalaga) [reply tooling/create-extension-templates/parent-pom.xml]: > having a structure similar to the the final directory layout could help a little

That's probably doable if the config of the `create-extension` mojo is adjusted too.

**Pull request merged**